packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { CSSProperties } from 'react';
+import DataTable from '@airbnb/lunar/lib/components/DataTable';
+import { Renderers, RendererProps } from '@airbnb/lunar/lib/components/DataTable/types';
+import { HEIGHT_TO_PX } from '@airbnb/lunar/lib/components/DataTable/constants';
+import { FormDataMetric, Metric } from '@superset-ui/chart';
+import { getNumberFormatter, NumberFormats } from '@superset-ui/number-format';
+import { getTimeFormatter } from '@superset-ui/time-format';
+
+type ColumnType = {
+  key: string;
+  label: string;
+  format: string;
+};
+
+type Props = {
+  data: any[];
+  height: number;
+  alignPositiveNegative?: boolean;
+  colorPositiveNegative?: boolean;
+  columns: ColumnType[];
+  filters?: {
+    [key: string]: any[];
+  };
+  includeSearch?: boolean;
+  metrics: FormDataMetric[];
+  onAddFilter?: (key: string, value: number[]) => void;
+  onRemoveFilter?: (key: string, value: number[]) => void;
+  orderDesc: boolean;
+  pageLength: number | string;
+  percentMetrics: string[];
+  tableFilter: boolean;
+  tableTimestampFormat: string;
+  timeseriesLimitMetric: FormDataMetric;
+};
+
+function NOOP(key: string, value: []) {}
+
+const defaultProps = {
+  alignPositiveNegative: false,
+  colorPositiveNegative: false,
+  filters: {},
+  includeSearch: false,
+  onAddFilter: NOOP,
+  onRemoveFilter: NOOP,
+};
+
+export type TableProps = Props & Readonly<typeof defaultProps>;
+
+type Cell = {
+  key: string;
+  value: any;
+};
+
+type TableState = {
+  selectedCells: Set<string>;
+};
+
+const NEGATIVE_COLOR = '#ff8787';
+const POSITIVE_COLOR = '#ced4da';
+
+const formatPercent = getNumberFormatter(NumberFormats.PERCENT_3_POINT);
+
+class TableVis extends React.Component<TableProps, TableState> {
+  static defaultProps = defaultProps;
+
+  constructor(props: TableProps) {
+    super(props);
+    this.state = {
+      selectedCells: new Set(),
+    };
+  }
+
+  handleCellSelected = (cell: Cell) => () => {
+    const { selectedCells } = this.state;
+    const { tableFilter, onRemoveFilter, onAddFilter } = this.props;
+
+    if (!tableFilter) {
+      return;
+    }
+    const newSelectedCells = new Set(Array.from(selectedCells));
+    const cellHash = `${cell.key}#${cell.value}`;
+    if (newSelectedCells.has(cellHash)) {
+      newSelectedCells.delete(cellHash);
+      onRemoveFilter(cell.key, [cell.value]);
+    } else {
+      newSelectedCells.add(cellHash);
+      onAddFilter(cell.key, [cell.value]);
+    }
+    this.setState({
+      selectedCells: newSelectedCells,
+    });
+  };
+
+  static getDerivedStateFromProps(props: TableProps, state: TableState) {
```

Comment:
Should we condition this update on a filter change?

Reply:
@williaster  I think this function should be always called whenever there is a prop change. Could you explain a bit more since I'm a bit confused 😆